sale_purchase_history/models/product_sale_history.py

```diff
@@ -0,0 +1,19 @@
+from odoo import api, fields, models
+
+class ProductSaleHistory(models.TransientModel):
+    _name = 'product.sale.history'
+
+    product_id = fields.Many2one('product.product')
+    is_sale = fields.Boolean(default=lambda self: self.env.context.get('is_sale', False))
+    is_purchase = fields.Boolean(default=lambda self: self.env.context.get('is_purchase', False))
+    sale_order_lines = fields.Many2many('sale.order.line', compute='_compute_previous_order_history', readonly=True)
+    purchase_order_lines = fields.Many2many('purchase.order.line', compute='_compute_previous_order_history', readonly=True)
```

Comment:
compute fields are readonly by default

Reply:
On a transient model wizard, the lines should be a new transient model as well.

With transient models, the wizard records are not created on the database. Therefore, you generally don't want to have the lines of a transient model to be of a model that is actually stored on the database, especially if they are editable.

Generally speaking, if it is for viewing purposes only, it is safe, but sometimes you might get unexpected results.